### DISCUSSION TITLE
MovieTutorial Genre

### DISCUSSION THREAD

**Discussion by Habib-Abdal** [serenity-is/Serenity#7191] [Help]
I have the problem which is saying "Invalid GenreId" the code and image of error is here as below:
MovieRow Code:
 `using Serenity.ComponentModel;
using Serenity.Data;
using Serenity.Data.Mapping;
using System;
using System.ComponentModel;

namespace MovieTutorial.MovieDb;

[ConnectionKey("Default"), Module("MovieDb"), TableName("[Mov].[Movie]")]
[DisplayName("Movies"), InstanceName("Movie")]
[ReadPermission("Administration:General")]
[ModifyPermission("Administration:General")]
[ServiceLookupPermission("Administration:General")]
public sealed class MovieTableRow : Row<MovieTableRow.RowFields>, IIdRow, INameRow
{
    [DisplayName("Movie Id"), Identity, IdProperty]
    public int? MovieId { get => fields.MovieId[this]; set => fields.MovieId[this] = value; }

    [DisplayName("Title"), Size(200), NotNull, QuickSearch, NameProperty]
    public string Title { get => fields.Title[this]; set => fields.Title[this] = value; }

    [DisplayName("Description"), Size(1000), QuickSearch(SearchType.StartsWith)]
    public string Description { get => fields.Description[this]; set => fields.Description[this] = value; }

    [DisplayName("Story"), QuickSearch]
    public string Storyline { get => fields.Storyline[this]; set => fields.Storyline[this] = value; }

    [DisplayName("Year"), QuickSearch(SearchType.Equals, numericOnly: 1)]
    public int? Year { get => fields.Year[this]; set => fields.Year[this] = value; }

    [DisplayName("Release Date"), QuickSearch]
    public DateTime? ReleaseDate { get => fields.ReleaseDate[this]; set => fields.ReleaseDate[this] = value; }

    [DisplayName("Kind"), NotNull, DefaultValue(MovieKind.Film), /*QuickSearch(searchType: SearchType.StartsWith)*/]
    public MovieKind? Kind { get => fields.Kind[this]; set => fields.Kind[this] = value; }

    [DisplayName("Runtime(mins)")]
    public int? Runtime {
        get => fields.Runtime[this];
        set => fields.Runtime[this] = value; }
    //adding genreId to movie Row
    [DisplayName("Genre"), ForeignKey("[Mov].Genre", "GenreId"), LeftJoin("jGenre")]
    [LookupEditor("MovieDb.Genre")]
    public Int32? GenreId { get  { return fields.GenreId[this]; } set { fields.GenreId[this] = value; } }

    [DisplayName("Genre"), Expression("jGenre.Name")]
    public String GenreName { get {  return fields.GenreName[this]; } set { fields.GenreName[this] = value; } }

    public class RowFields : RowFieldsBase
    {
        public Int32Field MovieId;
        public StringField Title;
        public StringField Description;
        public StringField Storyline;
        public Int32Field Year;
        public DateTimeField ReleaseDate;
        public Int32Field Runtime;
        public EnumField<MovieKind> Kind;
        public Int32Field GenreId;
        public StringField GenreName;
    
   
    }
}
`
Movie Form Code:   `using Serenity.ComponentModel;
using System;

namespace MovieTutorial.MovieDb.Forms;

[FormScript("MovieDb.MovieTable")]
[BasedOnRow(typeof(MovieTableRow), CheckNames = true)]
public class MovieTableForm
{

    public string Title { get; set; }
    [TextAreaEditor(Rows = 4)]
    public string Description { get; set; }
    [TextAreaEditor(Rows = 4)]
    public string Storyline { get; set; }
    public int Year { get; set; }
    public DateTime ReleaseDate { get; set; }

    public Int32 GenreId {  get; set; }
    public MovieKind Kind { get; set; }
    public int Runtime { get; set; }
    
}`
 the Genre Code:
 `using Serenity.ComponentModel;
using Serenity.Data;
using Serenity.Data.Mapping;
using System.ComponentModel;

namespace MovieTutorial.MovieDb;

[ConnectionKey("Default"), Module("MovieDb"), TableName("[Mov].[Genre]")]
[DisplayName("Genre"), InstanceName("Genre")]
[ReadPermission("Administration:General")]
[ModifyPermission("Administration:General")]
[ServiceLookupPermission("Administration:General")]
[LookupScript("MovieDb.Genre")]
public sealed class GenreRow : Row<GenreRow.RowFields>, IIdRow, INameRow
{
    [DisplayName("Genre Id"), Identity, IdProperty]
    public int? GenreId { get => fields.GenreId[this]; set => fields.GenreId[this] = value; }

    [DisplayName("Name"), Size(100), NotNull, QuickSearch, NameProperty]
    public string Name { get => fields.Name[this]; set => fields.Name[this] = value; }

    public class RowFields : RowFieldsBase
    {
        public Int32Field GenreId;
        public StringField Name;

    }
}`
the error: 
![image](https://github.com/serenity-is/Serenity/assets/136240124/074d90f9-28b1-4294-9f20-bee56293eae5)
![Screenshot 2024-06-29 192403](https://github.com/serenity-is/Serenity/assets/136240124/2f310a9c-96d7-425d-9b9a-b9ed985fe239)
please any one know the problem help me to solve it:
thank you

**Comment by VictorTomaili** [comment]
https://serenity.is/docs/tutorials/movies/07-adding-movie-genres

in the first step, you need to add migration for GenreId field.
